python/private/pypi/whl_library.bzl

```diff
@@ -295,12 +295,29 @@ def _whl_library_impl(rctx):
     elif rctx.attr.urls and rctx.attr.filename:
         filename = rctx.attr.filename
         urls = rctx.attr.urls
-        result = rctx.download(
-            url = urls,
-            output = filename,
-            sha256 = rctx.attr.sha256,
-            auth = get_auth(rctx, urls),
-        )
+        if len(urls) == 1 and not urls[0].startswith("https://"):
+            local_url = urls[0]
+            local_path = rctx.path(local_url)
```

Comment:
Ideally, the `whl_library` should consume locally available paths through labels, because then it makes things easier to thread through and `whl_library` already accepts a label. Would it be possible to create a `repository_rule` that exposes these paths as labels?